### PR TITLE
Add CI build config

### PR DESCRIPTION
**Pull request by ddorgan**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    - ./target/
+    - ./.cargo/
+
+
+#### stage:                        test
+
+test:rust:stable:
+  stage:                           test
+  script:
+    - time cargo test --all --release
```
Comment (pepyakin) [inline .gitlab-ci.yml]: I would rather go without `--release`

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+variables:
+  CI_SERVER_NAME:                  "GitLab CI"
+  CARGO_HOME:                      "${CI_PROJECT_DIR}/.cargo"
+
+
+cache:
+  key:                             "${CI_JOB_NAME}"
+  paths:
+    - ./target/
+    - ./.cargo/
```
Comment (pepyakin) [inline .gitlab-ci.yml]: Do we actually need a cache here?

File: .gitlab-ci.yml (hunk 1 above)
Comment (ddorgan) [reply .gitlab-ci.yml]: Removed --release

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  CARGO_HOME:                      "${CI_PROJECT_DIR}/.cargo"
+
+#### stage:                        test
+
+test:rust:stable:
+  stage:                           test
+  script:
+    - time cargo test --all
+  tags:
+    - rust-stable
```
Comment (bkchr) [inline .gitlab-ci.yml]: What does this tag do? Above you specify `rust:nightly` as image and the tag says stable?

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+
+#### stage:                        build
+
+build:linux:ubuntu:amd64:
+  stage:                           build
+  script:
+    - cargo build
+  tags:
+    - rust-stable
```
Comment (bkchr) [inline .gitlab-ci.yml]: Same as above.

File: .gitlab-ci.yml (hunk 3 above)
Comment (ddorgan) [reply .gitlab-ci.yml]: This image actually has more than one toolchain:

```
# rustup toolchain list
stable-x86_64-unknown-linux-gnu (default)
beta-x86_64-unknown-linux-gnu
nightly-x86_64-unknown-linux-gnu
```

We only use the tag to assign to a gitlab-runner. But maybe we should just use the tag rust instead. Will change.

**Pull request merged**